cs_tools/cli/tools/_scriptability/app.py

```diff
@@ -0,0 +1,363 @@
+# DEV NOTE:
+#
+# Future enhancements:
+#   - add tags on upload
+#   - add support for GUID mapping across instances of ThoughtSpot
+#   - ability to manipulate objects, such as renaming references to worksheets and tables, token replacement, etc.
+#   - add support for uploading .zip files
+#   - add support for tables and connections
+#
+
+import datetime
+import json
+import pathlib
+import click
+# import requests   # use httpx
+from thoughtspot_tml import YAMLTML
+from thoughtspot_tml.tml import TML
+from typer import Argument as A_, Option as O_  # noqa
+from typing import Dict, List, Optional
+import typer
+from zipfile import ZipFile
+
+from cs_tools import util
+from cs_tools.data.enums import GUID, TMLImportPolicy, TMLType, TMLContentType
+from cs_tools.api.middlewares import ConnectionMiddleware
+from cs_tools.cli.ux import _csv, console, CSToolsCommand, CSToolsGroup
+from cs_tools.cli.tools import common
+from cs_tools.cli.util import base64_to_file
+from cs_tools.thoughtspot import ThoughtSpot
+from cs_tools.cli.dependency import depends
+from cs_tools.cli.options import CONFIG_OPT, VERBOSE_OPT, TEMP_DIR_OPT
+
+
+def strip_blanks(inp: List[str]) -> List[str]:
+    return [e for e in inp if e]
+
+
+app = typer.Typer(
+    help="""
+    Tool for easily migrating TML between clusters.
+
+    ThoughtSpot provides the ability to extract object metadata (tables, worksheets, liveboards, etc.) 
+    in ThoughtSpot Modeling Language (TML) format, which is a text format based on YAML.  These files 
+    can be modified and uploaded into another instance. These files can then be modified and imported 
+    into another (or the same) instance to either create or modify objects.
+
+      cs_tools tools scriptability-migration --help
+    
+    \f
+    TODO - add more details on using the tools and a workflow.
+    
+    DEV NOTE:
+
+      Two control characters are offered in order to help with
+      docstrings in typer App helptext and command helptext
+      (docstrings).
+
+      \b - Preserve Whitespace / formatting.
+      \f - EOF, don't include anything after this character in helptext.
+    """,
+    cls=CSToolsGroup,
+    options_metavar='[--version, --help]'
+)
+
+
+@app.command(cls=CSToolsCommand)
+@depends(
+    thoughtspot=common.setup_thoughtspot,
+    options=[CONFIG_OPT, VERBOSE_OPT, TEMP_DIR_OPT],
+    enter_exit=True
+)
+def export(
```

Comment:
for `callback=_csv` .. are you envisioning users will use cli like this?

`cs_tools tools scriptability export --tags tag1,tag2,tag3 --export_ids guid1,guid2,guid3`

i created a new semantic for that will help us transition once typer passes through click's ability to support arbitrary types. can you see if using `CommaSeparatedValuesType` works here?

https://github.com/thoughtspot/cs_tools/blob/2835afc5a55b3ac9bcd9c578201dbc38caa115fc/cs_tools/cli/ux.py#L33-L45

impl. example here

https://github.com/thoughtspot/cs_tools/blob/2835afc5a55b3ac9bcd9c578201dbc38caa115fc/cs_tools/cli/tools/archiver/app.py#L73-L77

Reply:
I much prefer --tags tag1, tag2, tag3.  Updated to use the new type.